test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/MvcEndpointDataSourceTests.cs

```diff
@@ -152,16 +152,22 @@ private static TheoryData GetSingleActionData(bool isConventionalRouting)
                     {"{controller}/{id?}", isConventionalRouting ? new string[] { } : new[] { "TestController/{id?}" }},
                     {"{action}/{id?}", isConventionalRouting ? new string[] { } : new[] { "TestAction/{id?}" }},
                     {"{Controller}/{Action}/{id?}", new[] { "TestController/TestAction/{id?}" }},
+                    {"{Controller}/{Action}/{id?}/{more?}", new[] { "TestController/TestAction/{id?}/{more?}" }},
                     {"{CONTROLLER}/{ACTION}/{id?}", new[] { "TestController/TestAction/{id?}" }},
                     {"{controller}/{action=TestAction}", new[] { "TestController", "TestController/TestAction" }},
-                    {"{controller}/{action=TestAction}/{id?}", new[] { "TestController", "TestController/TestAction/{id?}" }},
-                    {"{controller=TestController}/{action=TestAction}/{id?}", new[] { "", "TestController", "TestController/TestAction/{id?}" }},
+                    {"{controller}/{action=TestAction}/{id?}", new[] { "TestController/TestAction/{id}", "TestController", "TestController/TestAction" }},
+                    {"{controller}/{action=TestAction}/{id?}/{more}", new[] { "TestController/TestAction/{id?}/{more}" }},
+                    {"{controller=TestController}/{action=TestAction}/{id?}", new[] { "TestController/TestAction/{id}", "", "TestController", "TestController/TestAction" }},
+                    {"{controller=TestController}/{action=TestAction}/{id?}/{more?}", new[] { "TestController/TestAction/{id}/{more}", "TestController/TestAction/{id}", "", "TestController", "TestController/TestAction" }},
                     {"{controller}/{action}/{*catchAll}", new[] { "TestController/TestAction/{*catchAll}" }},
                     {"{controller}/{action=TestAction}/{*catchAll}", new[] { "TestController", "TestController/TestAction/{*catchAll}" }},
-                    {"{controller}/{action=TestAction}/{id?}/{*catchAll}", new[] { "TestController", "TestController/TestAction/{id?}/{*catchAll}" }},
+                    {"{controller}/{action=TestAction}/{id?}/{*catchAll}", new[] { "TestController/TestAction/{id}/{*catchAll}", "TestController", "TestController/TestAction" }},
```

Comment:
This seems wrong? I expect to see `TestController/TestAction/{id}` also

Reply:
It isn't needed because `{*catchAll}` is optional.

Reply:
Ah - well spotted. Yeah I think you're right about this.